### PR TITLE
Add get_history() feature

### PR DESCRIPTION
**Pull request by aykim17**
I added a feature to pull a user's Reverso Context search history. I found it useful and thought others might as well.

### CODE REVIEW

File: reverso_context_api/client.py [hunk 1]
```diff
@@ -74,6 +74,17 @@ def get_favorites(self, source_lang=None, target_lang=None, cleanup=True):
...
+    def get_history(self, source_lang=None, target_lang=None):
+        """
+        Yields your search history (you have to provide login credentials to Client to use it)
+        :param source_lang: string of lang abbreviations separated by comma
+        :param target_lang: same as source_lang
+        :return: dict of sample attrs (source/target lang/text)
+        """
+        for page in self._history_pager(source_lang, target_lang):
+            for entry in page["results"]:
+                yield self._process_history_entry(entry, cleanup)
```
Comment (flagist0) [inline reverso_context_api/client.py]: `cleanup` is not defined in this context

File: reverso_context_api/client.py [hunk 2]
```diff
@@ -179,6 +217,21 @@ def _process_fav_entry(self, entry, cleanup):
             processed_entry[field_to] = val
         return processed_entry
 
+    def _process_history_entry(self, entry, cleanup):
```
Comment (flagist0) [inline reverso_context_api/client.py]: `cleanup` is not used by the code, so it's better to remove it

File: reverso_context_api/client.py [hunk 3]
```diff
@@ -179,6 +217,21 @@ def _process_fav_entry(self, entry, cleanup):
...
+        entry_fields_map = {
+            "srcLang": "source_lang",
+            "srcText": "source_text",
+            "trgLang": "target_lang",
+            "translation1": "translation_1",
```
Comment (flagist0) [inline reverso_context_api/client.py]: All translation variants are returned in fields "translationN". There can be many of them, I think they have to be extracted and put into list field "translations". Also sometimes there are even empty values, which have to be removed.

**Review by flagist0 — CHANGES_REQUESTED** [review]
Thank you for the pull request! I found couple of problems in the code. Unfortunately one of them doesn't let the code execute

**Comment by flagist0** [comment]
Ok, I've fixed the problems

**Comment by aykim17** [comment]
Sorry for the late reply and buggy submission- glad you were able to get it working!

**Pull request merged**